contracts/RollupBurnAuction.sol

```diff
@@ -0,0 +1,135 @@
+
+    address public forger;
+    Auction winner;
+    Auction auction;
+
+    /**
+     * @dev Event called when an operator beat the bestBid of the auction
+     */
+    event newBestBid(uint32 slot, address operator, uint128 amount);
+
+    /**
+     * @dev RollupBurnAuction constructor
+     * Set first block where the era will begin
+     * Initializes raffle for first era
+     * @param _rollup rollup main smart contract address
+     */
+    constructor(address _rollup) public {
+        require(_rollup != address(0), 'Address 0 inserted');
+        rollupInterface = RollupInterface(_rollup);
+        genesisBlock = getBlockNumber() + DELTA_BLOCKS_INITIAL_SLOT;
+        // Initialize first auction
+        forger = address(44);
+        setAuction(winner);
+        setAuction(auction);
+    }
+
+    function bid() external payable {
+        // if the ongoing auction has already finished, update it
+        updateAuction();
+        require(
+            msg.value > auction.bestBid.amount,
+            "Your bid doesn't beat the current best"
+        );
+        // refund, check 0 case (it means no bids yet for the auction, so no refund)
+        // DISCUSSION: should we refund?
+        if (auction.bestBid.amount > 0) {
+            auction.bestBid.operator.transfer(auction.bestBid.amount);
+            // emit refundBid(auction.slot, msg.sender, uint128(msg.value));
+        }
+        // set new best bider
+        auction.bestBid.operator = msg.sender;
+        auction.bestBid.amount = uint128(msg.value);
+        emit newBestBid(auction.slot, msg.sender, uint128(msg.value));
+    }
+
+    function forgeBatch(
+        address payable beneficiaryAddress,
+        uint[2] calldata proofA,
+        uint[2][2] calldata proofB,
+        uint[2] calldata proofC,
+        uint[8] calldata input
+     ) external {
+        // if the ongoing auction has already finished, update it
+        updateAuction();
+        require(msg.sender == forger, "Not allowed to forge");
+        rollupInterface.forgeBatch(beneficiaryAddress, proofA, proofB, proofC, input);
+    }
+
+    /**
+     * @dev Calculate slot from block number
+     * @param numBlock block number
+     * @return slot number
+     */
+    function block2slot(uint numBlock) public view returns (uint32) {
+        if (numBlock < genesisBlock) return 0;
+        return uint32((numBlock - genesisBlock) / (BLOCKS_PER_SLOT));
+    }
+
+    /**
+     * @dev Retrieve current slot
+     * @return slot number
+     */
+    function currentSlot() public view returns (uint32) {
+        return block2slot(getBlockNumber());
+    }
+
+    function setAuction(Auction storage _auction) internal {
+        _auction.slot = currentSlot() + 2;
+        _auction.bestBid.operator = address(44);
+        _auction.bestBid.amount = 0;
+    }
+
+    function updateAuction() internal {
+        // update forger
+        // DISCUSSION: if there is a period without bids or something goes wrong forger will stay
```

Comment:
I would restart auction for each slot not allowing a bidder to forge a batch beyond the slot he won.
This can be achieved better with `mapping (uint32 => Auction) winners;`